include/OpenSpaceToolkit/Astrodynamics/RootSolver.hpp

```diff
@@ -0,0 +1,113 @@
+/// Apache License 2.0
+
+#ifndef __OpenSpaceToolkit_Astrodynamics_RootSolver__
+#define __OpenSpaceToolkit_Astrodynamics_RootSolver__
+
+#include <boost/math/tools/roots.hpp>
+
+#include <OpenSpaceToolkit/Core/Types/Real.hpp>
+#include <OpenSpaceToolkit/Core/Types/Size.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+
+using ostk::core::types::Size;
+using ostk::core::types::Real;
+
+/// @brief                      A Root Solver is a tool for finding the roots of equations.
+class RootSolver
+{
+   public:
+    struct Solution
+    {
+        Real root;
+        Size numberOfIterations;
+    };
+
+    /// @brief              Constructor
+    ///
+    /// @code
+    ///                     RootSolver rootSolver = {aMaximumIterationsCount, aNumberOfDigits};
+    /// @endcode
+    ///
+    /// @param              [in] aMaximumIterationsCount The maximum iterations count for the solver
+    /// @param              [in] aNumberOfDigits The number of digits
+
+    RootSolver(const Size& aMaximumIterationsCount, const Size& aNumberOfDigits);
+
+    /// @brief              Virtual destructor
+
+    virtual ~RootSolver();
+
+    /// @brief              Output stream operator
+    ///
+    /// @param              [in] anOutputStream An output stream
+    /// @param              [in] aRootSolver A Root Solver
+    /// @return             An output stream
+
+    friend std::ostream& operator<<(std::ostream& anOutputStream, const RootSolver& aRootSolver);
+
+    /// @brief              Print root solver
+    ///
+    /// @param              [in] anOutputStream An output stream
+    /// @param              [in] (optional) displayDecorators If true, display decorators
+
+    void print(std::ostream& anOutputStream, bool displayDecorator = true) const;
+
+    /// @brief              Get maximum iterations count
+    ///
+    /// @return             Maximum iterations count
+
+    Size getMaximumIterationsCount() const;
+
+    /// @brief              Get number of digits of precision
+    ///
+    /// @return             Number of digits of precision
+
+    Size getNumberOfDigits() const;
+
+    /// @brief              Solve for root given a function, an initial guess and function direction
+    ///
+    /// @param              [in] aFunction A function
+    /// @param              [in] anInitialGuess An initial guess
+    /// @param              [in] isRising A boolean indicating whether the function is rising
+    ///
+    /// @return             The solution
+
+    Solution solve(
+        const std::function<double(const double&)>& aFunction, const double& anInitialGuess, const bool& isRising
```

Comment:
Just out of curiosity (let's not do that now), boost has some methods for multi-dimensional functions? see some good applications for multi-variate polynomials (laaaater)

Reply:
Interestingggg. I will do some research.